### PR TITLE
Clarify input to import_operator and import_state in doc string

### PR DESCRIPTION
**Pull request by Chiffafox**
**Context:**

The utility functions `import_state` and `import_operator` allow import of wavefunctions and operators in PennyLane from otuside sources (PySCF and OpenFermion, respectively).

**Description of the Change:**

Docstrings of `import_state` and `import_operator` are extended to clarify the currently supported sources of wavefunctions and operators, and the required input.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: doc/code/qml_qchem.rst [hunk 1]
```diff
@@ -13,7 +13,7 @@ number observables.
 .. automodapi:: pennylane.qchem
     :no-heading:
     :include-all-objects:
-    :skip: taper, symmetry_generators, paulix_ops, import_operator
+    :skip: taper, symmetry_generators, paulix_ops, import_operator, import_state
```
Comment (Jaybsoni) [inline doc/code/qml_qchem.rst:16]: Why was this needed? Just to be consistent with `import_operator?`

File: pennylane/qchem/convert.py [hunk 2]
```diff
@@ -564,8 +565,12 @@ def _ucisd_state(cisd_solver, tol=1e-15):
 def import_state(solver, tol=1e-15):
     r"""Convert an external wavefunction to a state vector.
 
+    Currently the only accepted source of wavefunctions is the PySCF library. The currently
+    supported methods are the restricted and unrestricted CISD/CCSD methods, with the function
+    accepting the corresponding Solver object from PySCF and inferring the type internally.
```
Comment (Jaybsoni) [inline pennylane/qchem/convert.py:570]: Don't need to detail the inner workings of the function

```suggestion
    Currently the only accepted source of wavefunctions is the PySCF library. The currently
    supported methods are the restricted and unrestricted CISD/CCSD methods.
```

File: pennylane/qchem/convert.py [hunk 3]
```diff
@@ -316,7 +317,7 @@ def import_operator(qubit_observable, format="openfermion", wires=None, tol=1e01
             If ``None``, the identity map (e.g., ``0->0, 1->1, ...``) will be used.
         tol (float): Tolerance in `machine epsilon <https://numpy.org/doc/stable/reference/generated/numpy.real_if_close.html>`_
             for the imaginary part of the coefficients in ``qubit_observable``.
-            Coefficients with imaginary part less than 2.22e-16*tol are considered to be real.
+            Coefficients with imaginary part less than :math:`2.22e-16 \cdot \text{tol}` are considered to be real.
```
Comment (Jaybsoni) [inline pennylane/qchem/convert.py:320]:
```suggestion
            Coefficients with imaginary part less than :math:`2.22e^{-16} \cdot \text{tol}` are considered to be real.
```

Comment (Jaybsoni) [reply pennylane/qchem/convert.py:320]: For exponentiation

**Review by Jaybsoni — COMMENTED** [review]
Just a few comments,

File: doc/code/qml_qchem.rst (hunk 1 above)
Comment (Chiffafox) [reply doc/code/qml_qchem.rst:16]: Yes -- otherwise it shows up as a function when you go to qml.qchem in the API. And `import_operator` does not, so I went  for consistency -- I think utility functions are not supposed to show up there.

File: pennylane/qchem/convert.py (hunk 3 above)
Comment (Chiffafox) [reply pennylane/qchem/convert.py:320]: I think that would look like "e" to the power of -16, which is not proper scientific notation. I'll convert it to `2.22 \cdot 10^{-16}`.

Comment (Jaybsoni) [reply pennylane/qchem/convert.py]: 💯

File: doc/code/qml_qchem.rst (hunk 1 above)
Comment (Jaybsoni) [reply doc/code/qml_qchem.rst:16]: 👍🏼

**Review by Jaybsoni — APPROVED** [review]
Nice 💯

File: pennylane/qchem/convert.py [hunk 4]
```diff
@@ -303,7 +303,8 @@ def _openfermion_pennylane_equivalent(
 def import_operator(qubit_observable, format="openfermion", wires=None, tol=1e010):
     r"""Convert an external operator to a PennyLane operator.
 
-    The external format currently supported is openfermion.
+    The external format currently supported is openfermion: the function takes most types of
+    OpenFermion qubit operators, such as those corresponding to Pauli words and summations thereof.
```
Comment (DSGuala) [inline pennylane/qchem/convert.py:307]: Not a mandatory change, just if you think it improves clarity. Maybe this actually sounds worse :laughing: 
```suggestion
    We currently support [openfermion](https://quantumai.google/openfermion) operators: the function accepts most types of
    OpenFermion qubit operators, such as those corresponding to Pauli words and sums of Pauli words.
```

**Review by DSGuala — APPROVED** [review]
Looks good. I just left some comments about wording/clarity. Good to merge as soon as you take a look at these :+1:

File: pennylane/qchem/convert.py (hunk 4 above)
Comment (Chiffafox) [reply pennylane/qchem/convert.py:307]: I like it! I'll take active voice over passive any day haha
